### PR TITLE
feat: add disabling of default 'everyone' group access to template

### LINKED ISSUE #7658: New templates purposely with no permissions (RBAC)
A customer reported a suggestion to allow new templates have no RBAC (groups).

Existing state is the Everyone group is attached to any new template.

This is a very good enterprise suggestion akin to zero trust then add permissions later.

### PR DESCRIPTION
**Pull request by Emyrk**
Fixes https://github.com/coder/coder/issues/7658

# Wording

Checkbox is **disabled** if you do not have the enterprise Template RBAC entitlement.

Screenshot from 2023-06-12 13-43-53

Screenshot from 2023-06-12 13-15-36

### CODE REVIEW

File: site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx [hunk 1]
```diff
@@ -384,39 +387,86 @@ export const CreateTemplateForm: FC<CreateTemplateFormProps> = ({
         title="Operations"
```
Comment (ammario) [inline site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx:387]: This section should be called Permissions now IMO.

**Review by ammario — DISMISSED** [review]

**Review by BrunoQuaresma — APPROVED** [review]
FE looks good!

File: cli/templatecreate.go [hunk 2]
```diff
@@ -144,6 +146,12 @@ func (r *RootCmd) templateCreate() *clibase.Cmd {
 		},
 	}
 	cmd.Options = clibase.OptionSet{
+		{
+			Flag: "disable-everyone",
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply cli/templatecreate.go:150]: That is a good idea, I had a hard time with the naming here.

File: site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx [hunk 3]
```diff
@@ -379,44 +382,91 @@ export const CreateTemplateForm: FC<CreateTemplateFormProps> = ({
...
+                />
+
+                <Stack direction="column" spacing={0.5}>
+                  <Stack
+                    direction="row"
+                    alignItems="center"
+                    spacing={0.5}
+                    className={styles.optionText}
+                  >
+                    <strong>Allow the everyone to use the template</strong>
```
Comment (deansheather) [inline site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx:442]:
```suggestion
                    <strong>Allow everyone to use the template</strong>
```

File: site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx [hunk 4]
```diff
@@ -379,44 +382,91 @@ export const CreateTemplateForm: FC<CreateTemplateFormProps> = ({
...
+                      </HelpTooltipText>
+                    </HelpTooltip>
+                  </Stack>
+                  <span className={styles.optionHelperText}>
+                    This setting requires an enterprise license for the&nbsp;
+                    <Link href="https://coder.com/docs/v2/latest/admin/rbac">
+                      &apos;Template RBAC&apos;
+                    </Link>{" "}
+                    feature to customize permissions. Without this license,
+                    unchecking this box prevents template usage by non-admins.
```
Comment (deansheather) [inline site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx:463]: They can't uncheck the box if they don't have the entitlement though, so this last sentence seems strange

**Review by deansheather — APPROVED** [review]
Go LGTM

File: site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx (hunk 4 above)
Comment (Emyrk) [reply site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx:463]: Fair. I was trying to word the rational why the checkbox was disabled. I'll just delete the extra sentence

**Pull request merged**